### ISSUE TITLE
webpack 5 incompatible unless configuring fullyResolved of false

### ISSUE THREAD

**Issue by bdefore** [Dan503/time-input-polyfill#35]
When upgrading from Webpack 4 to 5, I received a number of errors from this library

```
ERROR in ./node_modules/time-input-polyfill/index.mjs 1:0-56
Module not found: Error: Can't resolve './core/setters/apply_default' in '/Users/REDACTED/code/REDACTED/frontend/node_modules/time-input-polyfill'
Did you mean 'apply_default.js'?
BREAKING CHANGE: The request './core/setters/apply_default' failed to resolve only because it was resolved as fully specified
(probably because the origin is a '*.mjs' file or a '*.js' file where the package.json contains '"type": "module"').
The extension in the request is mandatory for it to be fully specified.
Add the extension to the request.
```

Other files that had errors (abbreviated, in general similar to above): 

```
Module not found: Error: Can't resolve './core/setters/update_time' in '/Users/REDACTED/code/REDACTED/frontend/node_modules/time-input-polyfill'
Did you mean 'update_time.js'?

Module not found: Error: Can't resolve './core/setters/set_data_attribute' in '/Users/REDACTED/code/REDACTED/frontend/node_modules/time-input-polyfill'
Did you mean 'set_data_attribute.js'?

Module not found: Error: Can't resolve './core/events/bind_events' in '/Users/REDACTED/code/REDACTED/frontend/node_modules/time-input-polyfill'
Did you mean 'bind_events.js'?

Module not found: Error: Can't resolve './core/setters/switch_times' in '/Users/REDACTED/code/REDACTED/frontend/node_modules/time-input-polyfill'
Did you mean 'switch_times.js'?

Module not found: Error: Can't resolve './core/getters/get_label' in '/Users/REDACTED/code/REDACTED/frontend/node_modules/time-input-polyfill'
Did you mean 'get_label.js'?

Module not found: Error: Can't resolve './core/accessibility/create_a11y_block' in '/Users/REDACTED/code/REDACTED/frontend/node_modules/time-input-polyfill'
Did you mean 'create_a11y_block.js'?
```

For now, this workaround was successful for me on the latest 1.0.10: https://github.com/graphql/graphql-js/issues/2721#issuecomment-723008284

Note that upgrading from graphql 14 -> 15 removes the requirement to add `fullyResolved: false` to Webpack config. I believe a similar fix could be made to this library by following the Webpack suggestion to add explicit extensions to the imports.

**Comment by Dan503** [comment]
Damn ☹

It looks like all I need to do is add `.js` to the end of all the import statements though to make it compatible.

I might be able to get around to this on the weekend.

**Comment by bdefore** [comment]
sorry, may be a false alarm, if per the docs i import from `time-input-polyfill/auto` instead of `time-input-polyfill` the error does not occur.

**Comment by Dan503** [comment]
@bdefore I've released a beta version for testing.

Can you try installing `npm i time-input-polyfill@beta` and seeing if you can still replicate the issue?

> sorry, may be a false alarm, if per the docs i import from `time-input-polyfill/auto` instead of `time-input-polyfill` the error does not occur.

I'm not sure why webpack was complaining about `time-input-polyfill` but not `time-input-polyfill/auto` :/

Anyway, `import TimePolyfill from 'time-input-polyfill'` is a valid thing you can do so I want to make sure it works.

**Comment by bdefore** [comment]
Thanks @Dan503 I can confirm that 1.0.11-beta.1 works for me without the `fullyResolved: false` workaround that 1.0.10 requires. I'm not sure I understand why this occurs. It feels unreasonable for consumer apps to force explicit extensions on library authors 🤷

**Comment by Dan503** [comment]
Thanks for testing that for me. I'll do a proper release when I get the chance.

I think it is because I was mixing `.mjs` with `.js` files and I wasn't being explicit with the imports about what file type to look for. The entry file was `.mjs` but the imports are all `.js`.

Though it seems like an odd requirement. If you can't find `.mjs` it would make sense to then look for a `.js` file 😕

**Comment by Dan503** [comment]
This should be fixed now in release 1.0.11
